src/common/crypto/mbedtls/cryptoprovider.cpp

```diff
@@ -286,27 +346,56 @@ aos::RetWithError<aos::SharedPtr<aos::crypto::PrivateKeyItf>> MbedTLSCryptoProvi
 aos::Error MbedTLSCryptoProvider::ASN1EncodeObjectIds(
     const aos::Array<aos::crypto::asn1::ObjectIdentifier>& src, aos::Array<uint8_t>& asn1Value)
 {
-    (void)src;
-    (void)asn1Value;
+    asn1Value.Resize(asn1Value.MaxSize());
 
-    return aos::ErrorEnum::eNotSupported;
+    uint8_t* start = asn1Value.Get();
+    uint8_t* p     = asn1Value.Get() + asn1Value.Size();
+
+    int len = aos::crypto::ASN1EncodeObjectIds(src, &p, start);
+    if (len < 0) {
+        return ErrorEnum::eFailed;
+    }
+
+    CopyForward(p, asn1Value.Get(), len);
```

Comment:
use `memmove` instead of custom function

Reply:
fixed